src/base/net/downloadmanager.cpp

```diff
@@ -40,11 +40,13 @@
 #include <QNetworkReply>
 #include <QNetworkRequest>
 #include <QSslError>
+#include <QTimer>
 #include <QUrl>
 
 #include "base/global.h"
 #include "base/logger.h"
 #include "base/preferences.h"
+#include "base/rss/rss_session.h"
```

Comment:
As I see it, you went the wrong way and introduced an unacceptable dependency. RSS subsystem is a client of Download Manager, so Download Manager should not depend on RSS subsystem.
If you don't have any ideas how to fix it, I can help with that later.
The first thing that comes to mind is to set a delay during sequential service registration (via `registerSequentialService()`). This will also allow you to set a separate delay for each site, if a single one for everyone is not enough (by the way, why not do it initially?).

Reply:
After looking at this i think i get the issue with the dependency of Download Manager by RSS. I think using `registerSequentialService` is a great idea and can present flexibility for future growth and avoid the problematic dependency between RSS and Download Manager.

I think i could overload the function `registerSequentialService` and allow a second parameter as the delay time, then add a QHash with the delays linked to the ServiceID, a better way might be to change the ServiceID struct and add the delay to it.
Please let me know what you think so i can make another commit or a rebase to make those changes.